Expensify/App
# Changed Profile Picture save on click of Save button

### Details

- For ProfilePage we were saving it immediately after uploading, which was a bit inconsistent from Workspace. To have a uniform user experience, with this PR, now it'll update the profile picture on click of the Save button.


### Fixed Issues

$ https://github.com/Expensify/App/issues/7015

### Tests
1. Uploading a new profile photo
2. Save button being disabled if the picture is not updated
3. Removing the existing profile picture
4. Reloading the screens without clicking Save (or going back)



### QA Steps

1. Go to the settings and click on Avatar
2. In the sidebar click on the profile avatar
3. Click on the edit (or camera) icon of Avatar and upload the photo
4. Ensure it doesn't upload the photo
5. Click on Save and verify that the avatar is saved


### Tested On


### Screenshots

#### Web

https://user-images.githubusercontent.com/3069065/150633707-753c3acb-f6fd-4443-9b4a-21d2adcfacec.mov

#### Mobile Web

https://user-images.githubusercontent.com/3069065/150633752-774aab22-9f3f-41f3-a2e5-dfb0c2bc1a23.mp4

#### Desktop

https://user-images.githubusercontent.com/3069065/150633751-a50859aa-138a-4ff7-aeea-d938358576b1.mp4

#### iOS


https://user-images.githubusercontent.com/3069065/150634231-02011450-6d1c-452d-9f59-db6192b8e6a3.mp4

#### Android


https://user-images.githubusercontent.com/3069065/150633743-dd36263e-3cf4-498d-8bff-494ae55157e9.mp4
